Sources/Orders/Models/OrderDataModel.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  OrderDataModel.swift
+//  PassKit
+//
+//  Created by Francesco Paolo Severino on 30/06/24.
+//
+
+import FluentKit
+
+/// Represents the `Model` that stores custom app data associated to Wallet orders.
+public protocol OrderDataModel: Model {
+    associatedtype OrderType: OrderModel
+
+    /// The foreign key to the order table
+    var order: OrderType { get set }
+}
+
+internal extension OrderDataModel {
+    var _$order: Parent<OrderType> {
+        guard let mirror = Mirror(reflecting: self).descendant("_order"),
+            let order = mirror as? Parent<OrderType> else {
+                fatalError("order property must be declared using @Parent")
```

Comment:
Not a big fan of these `fatalError`s honestly

Reply:
What could I replace them with? In most cases I cannot make these properties optional